tutor/src/components/annotations/annotation.jsx

```diff
@@ -347,64 +338,52 @@ export default class AnnotationWidget extends React.Component {
   @action.bound
   updateActiveAnnotation(event) {
     const newValue = event.target.value;
-
-    this.activeHighlight.annotation = newValue;
+    this.activeHighlight.text = newValue;
   }
 
   @autobind
-  updateAnnotation(entry) {
-    if (entry.lastSavedAnnotation !== entry.annotation) {
-      hypothesisStore.update(entry.savedId, entry.annotation).then(
-        action((response) => {
-          entry.lastSavedAnnotation = entry.annotation;
-          return entry;
-        }));
-    }
+  updateAnnotation(annotation) {
+    return annotation.save();
+    // debugger
+
+    // if (entry.lastSavedAnnotation !== entry.annotation) {
+    //   hypothesisStore.update(entry.savedId, entry.annotation).then(
+    //     action((response) => {
+    //       entry.lastSavedAnnotation = entry.annotation;
+    //       return entry;
+    //     }));
+    // }
   }
 
   @action.bound
   updateHighlightedAnnotation() {
-    const entry = this.activeHighlight;
-
+    const annotation = this.activeHighlight;
     this.props.windowImpl.getSelection().empty();
-    this.updateAnnotation(entry);
+    this.updateAnnotation(annotation);
     this.activeHighlight = null;
   }
 
   @autobind
-  saveHighlight() {
+  saveNewHighlight() {
     this.props.windowImpl.getSelection().empty();
-
-    return hypothesisStore.save(this.props.documentId, this.savedSelection, '',
-      {
-        ecosystemId: this.props.ecosystemId,
-        chapter: this.props.chapter,
-        section: this.props.section,
-        title: this.props.title
-      }
-    ).then(action((response) => {
-        // Have to make it observable before pushing it, or it won't be the same
-        // object as what is in the observable array
-        const newEntry = observable(dbEntryToPreload(response));
-
-        this.allAnnotationsForThisBook.push(newEntry);
-        return newEntry;
-      }));
+    return User.annotations.create({
+      documentId: this.props.documentId,
+      selection: this.savedSelection,
+      ecosystemId: this.props.ecosystemId,
```

Comment:
This is my fault, but ecosystemId in the annotation storage is actually courseId. That's necessary for the "open book in new tab" feature. I didn't get the variable names changed over.

Reply:
Ah ok. I'll update in the followup that adds the toggle